source/Coop/Mod/Persistence/EntityManager.cs

```diff
@@ -75,6 +75,11 @@ private void AddPendingParties(Tick tick)
 
         private void InitRoom(RailServerRoom room)
         {
+            if (Campaign.Current == null)
+            {
+                throw new Exception("Unable to initialize game entities: Unexpected state. No game loaded?");
+            }
+            // TODO: If the server runs in a separate thread we need to synchronize modifying state.
```

Comment:
There's `GameLoopRunner.RunOnMainThread` to access game state in the games main thread.

Reply:
Non issue removed